cdap-docs/deploy.sh

```diff
@@ -0,0 +1,165 @@
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+################################################################################
+# Deployment script for docs
+# Deploys zip files created by build scripts
+#
+# OPT_DIR is an optional directory to use instead of derived branch name. E.g.:
+#   if branch name=feature/docs-build-testing, and we want to use testdir
+#   we would set OPT_DIR=testdir and add it to the bamboo tasks' variables
+#   and the remote directory would be 2.8.1/testdir (testdir treated as branch)
+# If OPT_DIR is not set, bamboo.planRepository.<position>.branch will be used,
+#   e.g. /var/www/html/cdap/2.8.1/feature-myfix, unless the branch name
+#   is release/* or develop/* in which case, it will just be put into
+#   the version directory, e.g. /var/www/html/cdap/2.8.1
+#
+# function variables reference: (these were kept consistent throughout)
+# $1 user name (typically bamboo)
+# $2 remote server (docs or docs staging servers)
+# $3 remote web directory
+# $4 zip archive file
+# $5 local path to zip archive file
+################################################################################
+
+# output trimmer
+decho () {
+  if [[ "${DEBUG}" == 'yes' ]]; then
+    echo ${*}
+  else
+    RSYNC_QUIET='--quiet'
+  fi
+}
+
+# set the mandatory variables first (from info in project's main pom.xml)
+function get_version () {
+  TMP_VERSION=`grep "<version>" ../pom.xml`
+  TMP_VERSION=${TMP_VERSION#*<version>}
+  VERSION=${TMP_VERSION%%</version>*}
+}
+
+function get_project () {
+  TMP_PROJECT=`grep "<artifactId>" ../pom.xml`
+  TMP_PROJECT=${TMP_PROJECT#*<artifactId>}
+  PROJECT=${TMP_PROJECT%%</artifactId>*}
+}
+
+function convert_branch_name () {
+  decho "converting '/' to '-' in branch name"
+  DOC_DIR=`echo $DOC_DIR | tr '/' '-'`
+}
+
+set_remote_dir () {
+  DOC_DIR=${OPT_DIR:-${BRANCH_NAME}}
+  convert_branch_name
+  if [[ "${DOC_DIR}" == release* || "${DOC_DIR}" == develop* ]]; then
+    DIR=''
+  else
+    DIR=branches/${DOC_DIR}
+  fi
+  decho "SUBDIR=${DIR}"
+}
+
+# parameters that can be passed to this script (as environment variables)
+DEBUG=${DEBUG:-no}
+DEPLOY_TO_STG=${DEPLOY_TO_STG:-no}
+DEPLOY_TO_DOCS=${DEPLOY_TO_DOCS:-no}
+BUILD_WORKING_DIR=${BUILD_WORKING_DIR:-/var/bamboo/xml-data/build-dir/CDAP-DRBD-JOB1}
+DOCS_SERVER1=${DOCS_SERVER1:-docs1.cask.co}
+DOCS_SERVER2=${DOCS_SERVER2:-docs2.cask.co}
+STG_SERVER=${STG_SERVER:-docs-staging.cask.co}
+REMOTE_STG_BASE=${REMOTE_STG_BASE:-/var/www/html/cdap}
+REMOTE_DOCS_BASE=${REMOTE_DOCS_BASE:-/var/www/docs/cdap}
```

Comment:
I grab "cdap" from the project's pom.xml (artifactId).  If we were ever to change CDAP's name to something else, this script would not have to be changed.